src/tonic/mod.rs

```diff
@@ -79,6 +80,9 @@ pub fn init(
             .layer(tonic_telemetry::TonicTelemetryLayer::new(
                 telemetry_collector,
             ))
+            .layer(api_key::ApiKeyMiddlewareLayer::new(
```

Comment:
Is it possible to use `option_layer` to only add the layer if the `api_key` is defined?

This way the middleware would always have an `api-key` to work with mirroring the REST middleware,